lalrpop/src/build/mod.rs

```diff
@@ -29,32 +31,56 @@ mod fake_term;
 
 use self::fake_term::FakeTerminal;
 
+pub fn process() -> io::Result<()> {
+    let session = Session::new();
+
+    let src_dir = try!(current_dir()).join("src");
+
+    // The environment variable OUT_DIR is set by cargo, and specifies
+    // a directory where generated code should be put.
+    process_dir(&session, &src_dir, Path::new(&env::var("OUT_DIR").unwrap()))
+}
+
 pub fn process_root() -> io::Result<()> {
     let session = Session::new();
-    process_dir(&session, try!(current_dir()))
+    let cwd = try!(current_dir());
+    process_dir(&session, &cwd, &cwd)
```

Comment:
Maybe; I just kept it like this for backwards compatibility.  There is the `.process()` function that does what you expect.

Note that putting things in `OUT_DIR` is *not* equivalent to putting it into the `src` dir.  You still need to `include!()` the generated file from a file that's checked in.  That's why changing `process_root()` to write its generated files to `OUT_DIR` would be a breaking change.